### PR TITLE
[do not merge] add math.copySign

### PR DESCRIPTION
**Pull request by timotheecour**
just to show how to use vmops for https://github.com/nim-lang/Nim/pull/16406

### CODE REVIEW

File: compiler/condsyms.nim [hunk 1]
```diff
@@ -123,3 +123,4 @@ proc initDefines*(symbols: StringTableRef) =
   defineSymbol("nimHasEffectTraitsModule")
   defineSymbol("nimHasCastPragmaBlocks")
   defineSymbol("nimHasDeclaredLocs")
+  defineSymbol("nimHasCopySign")
```
Comment (timotheecour) [inline compiler/condsyms.nim:126]: in this case, not actually needed

File: tests/stdlib/tmath.nim [hunk 2]
```diff
@@ -338,8 +338,11 @@ template main =
     doAssert copySign(10.0, 0.0) == 10.0
     doAssert copySign(-1.0, NaN) == 1.0
     doAssert copySign(10.0, NaN) == 10.0
-    doAssert copySign(-1.0, -NaN) == 1.0
-    doAssert copySign(10.0, -NaN) == 10.0
+
+    when nimvm: discard
+    else:
+      doAssert copySign(-1.0, -NaN) == 1.0
```
Comment (timotheecour) [inline tests/stdlib/tmath.nim:344]: this is a bug, we should track it (the fact that it gives different results for vm vs rt)

**Review by timotheecour — COMMENTED** [review]
misc

File: tests/stdlib/tmath.nim (hunk 2 above)
Comment (timotheecour) [reply tests/stdlib/tmath.nim:344]: EDIT: caused by https://github.com/timotheecour/Nim/issues/499 (you can ref this bug in your PR) /cc @xflywind

Comment (ringabout) [reply tests/stdlib/tmath.nim:344]: Ok

Comment (xflywind) [reply tests/stdlib/tmath.nim:344]: Why disable it? It works for me.

Comment (ringabout) [reply tests/stdlib/tmath.nim:344]: Why disable it? It works for me.

Comment (timotheecour) [reply tests/stdlib/tmath.nim:344]: maybe it's osx specific? what do you get for https://github.com/timotheecour/Nim/issues/499 ?

Comment (ringabout) [reply tests/stdlib/tmath.nim:344]: I don't know. It seems that It works with `koch temp c -r` but doesn't work with `nim c -r --lib:lib` and Nim CI> 
Yeah, I mean my `copySign` tests

Comment (ringabout) [reply tests/stdlib/tmath.nim:344]: > maybe it's osx specific? what do you get for #499 ?

#499 is not OS specific, I got the same result in windows

**Comment by timotheecour** [comment]
closing now that https://github.com/nim-lang/Nim/pull/16406 was merged

**Pull request closed without merging**